ext/fetch/23_request.js

```diff
@@ -592,8 +592,8 @@ const signalAbortError = new DOMException(
 );
 ObjectFreeze(signalAbortError);
 
-function abortRequest(request) {
-  if (request[_signal]) {
+function abortRequest(request, createNew = false) {
```

Comment:
Unfortunately this won't work either. Most likely we'll need to set some sort of semaphore in the signal cache to state that any signal created here needs to be aborted on creation.

Something like `if (request[_signalCache] === undefined) { request[_signalCache] = false } else { request[_signalCache][signalAbort](signalAbortError); }`. In the signal getter, we should check `request[_signalCache] === false`